src/avatar/Avatar.js

```diff
@@ -45,25 +49,10 @@ const Avatar = props => {
 
   let { width, height } = props;
 
-  if (small) {
-    width = 34;
-    height = 34;
-  } else if (medium) {
-    width = 50;
-    height = 50;
-  } else if (large) {
-    width = 75;
-    height = 75;
-  } else if (xlarge) {
-    width = 150;
-    height = 150;
-  } else if (!width && !height) {
-    width = 34;
-    height = 34;
-  } else if (!width) {
-    width = height;
-  } else if (!height) {
-    height = width;
+  const iconDimension = DEFAULT_SIZES[size];
+
+  if (iconDimension) {
+    [width, height] = iconDimension;
```

Comment:
I was just about to type this on the way to work :D Good job. Think we can set `small` as true by default.

Is there any reason we explicitly have `height` and `width` as props. If the avatar is either a circle or rectangle that means that width and height will always be the same no? If so `size` could either be `'small', 'medium', 'large', 'xlarge'` or a number.

Reply:
@Gregjarvez 
I think the idea of @iRoachie to have either a string or a number is cool.
You can just have something like `width = typeof props.size === 'number' ? props.size : DEFAULT_SIZES[size][0] || DEFAULT_SIZES.small[0]` (doesn't have to be exactly this but you got the idea), same goes for the `height`. Be sure you also set `small` as the `defaultProps` for `size`.

Reply:
Fantastic. so which props do we get rid of apart from `width and height`

Reply:
Oh was just talking about `'small', 'medium', 'large', 'xlarge', 'width', 'height'`. 6 become 1 props 🔥

Reply:
oh ok. I will get that done soon !!

Reply:
@iRoachie @xavier-villelegier I did the changes required seem to work 
https://snack.expo.io/rJ_bq9isz
changed code 
```javascript 
// iRoachie mentioned avatar is either a circle or a square. one number will work fine
const DEFAULT_SIZES = {
  small: 34,
  medium: 50,
  large: 75,
  xlarge: 150,
};
//size is `small` by default in defaultProps
 let { size } = props;

const iconDimension = typeof size === 'number' ? size : DEFAULT_SIZES[size];

let height;
let width = (height = iconDimension);

```

Reply:
@Gregjarvez Just add a fallback on the small size for `iconDimension` (`DEFAULT_SIZES[size] || DEFAULT_SIZES.small`) in case you typed an invalid string size

Reply:
Otherwise it looks good to me